test/mobile/test_lite_script_module.py

```diff
@@ -55,13 +55,89 @@ def forward(self, x):
+
+            def forward(self, x):
+                return x + 1
+
+        class B(torch.nn.Module):
+            def __init__(self):
+                super(B, self).__init__()
+                self.A0 = A()
+                self.A1 = A()
+
+            def forward(self, x):
+                return self.A0(x) + self.A1(x)
+
+        input_data = torch.tensor([5])
+        scripted_module = torch.jit.script(B(), input_data)
+        exported_module = scripted_module._save_to_buffer_for_lite_interpreter(_save_mobile_debug_info=True)
+
+        assert(b"mobile_debug.pkl" in exported_module)
+        assert(b"module_debug_info" in exported_module)
+        assert(b"top(B).forward" in exported_module)
+        assert(b"top(B).A0(A).forward" in exported_module)
+        assert(b"top(B).A1(A).forward" in exported_module)
 
-        assert(b"mobile_debug.pkl" in bytes)
-        assert(b"module_debug_info" in bytes)
-        assert(b"top(B).forward" in bytes)
-        assert(b"top(B).A0(A).forward" in bytes)
-        assert(b"top(B).A1(A).forward" in bytes)
+    def test_save_mobile_module_with_debug_info_with_script_nested_call(self):
+        class A(torch.nn.Module):
+            def __init__(self):
+                super(A, self).__init__()
+
+            def forward(self, x):
+                return x + 1
+
+        class B(torch.nn.Module):
+            def __init__(self):
+                super(B, self).__init__()
+
+            def forward(self, x):
+                return x + 2
+
+        class C(torch.nn.Module):
+            def __init__(self):
+                super(C, self).__init__()
+                self.A0 = A()
+                self.B0 = B()
+
+            def forward(self, x):
+                z = self.B0(x)
+                return self.A0(z)
+                # return self.A0(self.B0(x)) + 1
+
+        input = torch.tensor([5])
+        scripted_module = torch.jit.script(C(), input)
+        print("start")
+        print(scripted_module._c.dump_to_str())
+        print("end")
+
+        optimized_scripted_module = optimize_for_mobile(scripted_module)
+
+        exported_module = scripted_module._save_to_buffer_for_lite_interpreter(_save_mobile_debug_info=True)
+        optimized_exported_module = optimized_scripted_module._save_to_buffer_for_lite_interpreter(_save_mobile_debug_info=True)
+
+        assert(b"mobile_debug.pkl" in exported_module)
+        assert(b"module_debug_info" in exported_module)
+        assert(b"top(C).forward" in exported_module)
+        assert(b"top(C).A0(A).forward" in exported_module)
+        assert(b"top(C).A0(A).forward.B0(B).forward" in exported_module)
+
+        assert(b"mobile_debug.pkl" in optimized_exported_module)
+        assert(b"module_debug_info" in optimized_exported_module)
+        assert(b"top(C).forward" in optimized_exported_module)
+        assert(b"top(C).A0(A).forward" in optimized_exported_module)
+        assert(b"top(C).A0(A).forward.B0(B).forward" in optimized_exported_module)
```

Comment:
I don't think `top(C).A0(A).forward.B0(B).forward` should exist since the nesting doesn't seem to suggest this.

Reply:
Yes that's right. It is the issue I'm trying to address in the latest commit. After a more proper fix, I will fix this unittest.